Core/src/org/sleuthkit/autopsy/communications/AccountDeviceInstanceNode.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2017-18 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.communications;
+
+import java.awt.event.ActionEvent;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import javax.swing.AbstractAction;
+import javax.swing.Action;
+import org.openide.nodes.AbstractNode;
+import org.openide.nodes.Children;
+import org.openide.nodes.Sheet;
+import org.openide.util.NbBundle;
+import org.openide.util.Utilities;
+import org.openide.util.lookup.Lookups;
+import org.sleuthkit.autopsy.datamodel.NodeProperty;
+import org.sleuthkit.datamodel.Account;
+import org.sleuthkit.datamodel.AccountDeviceInstance;
+import org.sleuthkit.datamodel.CommunicationsFilter;
+import org.sleuthkit.datamodel.CommunicationsManager;
+
+/**
+ * Node to represent an Account Device Instance in the CVT
+ */
+final class AccountDeviceInstanceNode extends AbstractNode {
+
+    private final AccountDeviceInstanceKey accountDeviceInstanceKey;
+    private final CommunicationsManager commsManager;
+    private final Account account;
+
+    AccountDeviceInstanceNode(AccountDeviceInstanceKey accountDeviceInstanceKey, CommunicationsManager commsManager) {
+        super(Children.LEAF, Lookups.fixed(accountDeviceInstanceKey, commsManager));
+        this.accountDeviceInstanceKey = accountDeviceInstanceKey;
+        this.commsManager = commsManager;
+        this.account = accountDeviceInstanceKey.getAccountDeviceInstance().getAccount();
+        setName(account.getTypeSpecificID());
+        setIconBaseWithExtension("org/sleuthkit/autopsy/communications/images/" + Utils.getIconFileName(account.getAccountType()));
+    }
+
+    public AccountDeviceInstance getAccountDeviceInstance() {
+        return accountDeviceInstanceKey.getAccountDeviceInstance();
+    }
+
+    public AccountDeviceInstanceKey getAccountDeviceInstanceKey() {
+        return accountDeviceInstanceKey;
+    }
+
+    public CommunicationsManager getCommsManager() {
+        return commsManager;
+    }
```

Comment:
I am not sure I am comfortable with making a node class a source for getting the CommunicationsManager.  Is the motivation for this to avoid having to do Case.getCurrentCase().getSleuthkitCase().getCommunicationsManager(), or are there other reasons?

Reply:
that is exactly the motivation.  I particularly don't like calling get current case, as it always introduces a lot of noise to do the proper error handling.  However if you prefer that to the approach of passing he CommunicationsManager around with he nodes, I will change it.